lsp/nls/src/requests/completion.rs

```diff
@@ -167,33 +170,61 @@ fn remove_duplicates(items: &Vec<CompletionItem>) -> Vec<CompletionItem> {
             let (ty, _) = linearization.resolve_item_type_meta(item);
             match (&item.kind, ty) {
                 // Get record fields from static type info
-                (TermKind::Declaration(..), Types(TypeF::Record(rrows))) if id == item.id => {
-                    Some((find_fields_from_type(&rrows, path), item.ty.clone()))
+                (TermKind::Declaration(..), Types(TypeF::Record(row))) => {
+                    vec![(find_fields_from_type(&row, path), item.ty.clone())]
                 }
-                (TermKind::Declaration(_, _, ValueState::Known(body_id)), _) if id == item.id => {
-                    match find_fields_from_contract(&linearization, *body_id, path) {
-                        // Get record fields from contract metadata
-                        Some(fields) => Some((fields, item.ty.clone())),
-                        // Get record fields from lexical scoping
-                        None => find_fields_from_term_kind(&linearization, id, path)
-                            .map(|idents| (idents, item.ty.clone())),
-                    }
+                (TermKind::Declaration(_, _, ValueState::Known(body_id)), _) => {
+                    find_fields_from_contract(&linearization, *body_id, path)
+                        .or_else(|| find_fields_from_term_kind(&linearization, id, path))
+                        .map(|idents| vec![(idents, item.ty.clone())])
+                        .unwrap_or_default()
                 }
-                _ => None,
+                (
+                    TermKind::RecordField {
+                        value: ValueState::Known(value),
+                        ..
+                    },
+                    _,
+                ) => find_fields_from_term_kind(&linearization, *value, path)
+                    .map(|idents| vec![(idents, item.ty.clone())])
+                    .unwrap_or_default(),
+                _ => Vec::new(),
             }
         })
-        .collect()
+        .unwrap_or_default()
+}
+
+/// Generate completion for a stdlib identifier.
+fn stdlib_completion(
+    server: &Server,
+    name: Ident,
+    path: &mut Vec<Ident>,
+) -> Option<Vec<(Vec<Ident>, Types)>> {
+    let module = StdlibModule::try_from(name).ok()?;
+    let file_id = server.cache.get_submodule_file_id(module)?;
+    let lin = server.lin_cache_get(&file_id).ok()?;
+
+    // Stdlib modules are represented by a record with a single field which is named
+    // after the module name. Here, we try to get the ID of the
+    // `TermKind::RecordField {..}` associated with that field, which defines the module.
+    // Also, because of the way the linearization algorithm works, we can be almost sure that
+    // the ID of this field is `1`, which will give us constant time access to it, but
+    // this might not always be the case (the algorithm can change), so we have to settle
+    // for a linear search here.
+    let id = lin.linearization.iter().find_map(|item| match &item.kind {
```

Comment:
I don't think we can, because the environment of the current file doesn't tell us anything about the linearization of the stdlib (which is a different file). That means that we cannot get the `id` of the `array` field (in this case).

Another approach might be to look into the type-checking context, because the `stdlib` modules are already there, but this would be different from the normal completion mechanism  we already have